src/views/error-views/dev-dot-404/index.tsx

```diff
@@ -0,0 +1,78 @@
+import React, { ReactElement } from 'react'
+import { useRouter } from 'next/router'
+import { IconWaypointColor16 } from '@hashicorp/flight-icons/svg-react/waypoint-color-16'
+import { IconVaultColor16 } from '@hashicorp/flight-icons/svg-react/vault-color-16'
+import { IconHome16 } from '@hashicorp/flight-icons/svg-react/home-16'
+import { useErrorPageAnalytics } from '@hashicorp/react-error-view'
+import InlineLinkButton from 'components/inline-link-button'
+import CardsGridList from 'components/cards-grid-list'
+import IconCardLink, { IconCardLinkProps } from 'components/icon-card-link'
+import {
+  ErrorViewContainer,
+  ErrorViewH1,
+  ErrorViewParagraph,
+} from '../components'
+
+import s from './dev-dot-404.module.css'
+
+/**
+ * Generic 404 error view content for use in dev-dot.
+ */
+export function DevDot404(): ReactElement {
+  const router = useRouter()
+  useErrorPageAnalytics(404)
+
+  return (
+    <ErrorViewContainer>
+      <ErrorViewH1>
+        We couldn&apos;t find the page you&apos;re looking&nbsp;for.
+      </ErrorViewH1>
+      <ErrorViewParagraph>
+        Please check the url you entered for typos,{' '}
+        <InlineLinkButton onClick={router.back}>go back</InlineLinkButton> to
+        the page you came from, or go to one of the pages below.
+      </ErrorViewParagraph>
+      <div className={s.cards}>
+        <CardsGridList>
```

Comment:
💭 I think you may want to use `IconCardLinkGridList` here, which should reduce the length of this code a bit I think!

Reply:
Had tried initially, unfortunately `IconCardLinkGridList` supports only a top-level `productSlug`, rather than individual `productSlug` values per card.

I felt it was easier and almost as clean to compose things this way. But happy to update `IconCardLinkGridList` to support say `defaultProductSlug` at the top level, and optional `productSlug` properties per `IconCard` if this feels like a blocker!

Reply:
Ah ok this makes sense! ✨ 

🎨 I'll leave it up to you if you want to make a change here or not! I think we probably want to make the change at some point.

---

💭 This is what I'd probably do to make `IconCardLinkGridList` more reusable:

- Make `productSlug` optional
- Update the `IconCardLink`s rendered there to accept `productSlug || card.productSlug`

💭 Another possible optimization: make `IconCardLink`'s `productSlug` default to `hashicorp`.